src/Data/Text/Builder/Linear/Dec/Bounded.hs

```diff
@@ -98,16 +105,17 @@ unsafePrependDec marr !off n
     go o k
       | k >= 10 = do
           let (q, r) = quotRem100 k
-          A.copyFromPointer marr (o - 1) (Ptr digits `plusPtr` (fromIntegral r `shiftL` 1)) 2
+          A.copyFromPointer marr (o - 1) (digits `plusPtr` (fromIntegral r `shiftL` 1)) 2
           if k < 100 then pure (o - 1) else go (o - 2) q
       | otherwise = do
           A.unsafeWrite marr o (fromIntegral (0x30 + k))
           pure o
-
-    digits ∷ Addr#
-    digits = "00010203040506070809101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899"#
 {-# INLINEABLE unsafePrependDec #-}
 
+digits ∷ CString
+digits = Ptr "00010203040506070809101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899"#
+{-# NOINLINE digits #-}
```

Comment:
Doesn't it introduce a redundant boxing?

Reply:
No: the use site the unboxing and refers directly to the `Addr#` value. It is boxed here only because GHC does not accept unboxed values at top level.